internals/overlord/servstate/manager_test.go

```diff
@@ -984,31 +989,40 @@ func (s *S) TestOnCheckFailureRestartWhileRunning(c *C) {
 	s.manager.NotifyPlanChanged(checkMgr.PlanChanged)
 
 	// Tell service manager about check failures
-	checkMgr.NotifyCheckFailed(s.manager.CheckFailed)
+	checkFailed := make(chan int)
```

Comment:
Nit: normally when you don't care what the value is, you'd use a channel of empty struct (`chan struct{}`) to indicate that, and then send `checkFailed <- struct{}{}` below.

Reply:
IINM `chan struct{}` is the most memory-efficient.

Reply:
Thank you! Noted.